src/main/kotlin/no/nav/familie/ef/sak/mapper/IverksettingDtoMapper.kt

```diff
@@ -0,0 +1,168 @@
+import no.nav.familie.ef.sak.service.FagsakService
+import no.nav.familie.ef.sak.service.PersisterGrunnlagsdataService
+import no.nav.familie.ef.sak.service.SøknadService
+import no.nav.familie.ef.sak.service.TilkjentYtelseService
+import no.nav.familie.ef.sak.service.steg.StegType
+import no.nav.familie.kontrakter.ef.felles.BehandlingType
+import no.nav.familie.kontrakter.ef.felles.BehandlingÅrsak
+import no.nav.familie.kontrakter.ef.felles.StønadType
+import no.nav.familie.kontrakter.ef.iverksett.AdressebeskyttelseGradering
+import no.nav.familie.kontrakter.ef.iverksett.AndelTilkjentYtelseDto
+import no.nav.familie.kontrakter.ef.iverksett.BehandlingsdetaljerDto
+import no.nav.familie.kontrakter.ef.iverksett.DelvilkårsvurderingDto
+import no.nav.familie.kontrakter.ef.iverksett.FagsakdetaljerDto
+import no.nav.familie.kontrakter.ef.iverksett.InntektDto
+import no.nav.familie.kontrakter.ef.iverksett.IverksettDto
+import no.nav.familie.kontrakter.ef.iverksett.PeriodebeløpDto
+import no.nav.familie.kontrakter.ef.iverksett.Periodetype
+import no.nav.familie.kontrakter.ef.iverksett.SøkerDto
+import no.nav.familie.kontrakter.ef.iverksett.TilkjentYtelseDto
+import no.nav.familie.kontrakter.ef.iverksett.VedtaksdetaljerDto
+import no.nav.familie.kontrakter.ef.iverksett.VilkårsvurderingDto
+import no.nav.familie.kontrakter.ef.iverksett.VurderingDto
+import no.nav.familie.kontrakter.felles.annotasjoner.Improvement
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.util.UUID
+import no.nav.familie.kontrakter.ef.felles.RegelId as RegelIdIverksett
+import no.nav.familie.kontrakter.ef.felles.VilkårType as VilkårTypeIverksett
+import no.nav.familie.kontrakter.ef.felles.Vilkårsresultat as VilkårsresultatIverksett
+import no.nav.familie.kontrakter.ef.iverksett.SvarId as SvarIdIverksett
+
+@Component
+class IverksettingDtoMapper(private val arbeidsfordelingService: ArbeidsfordelingService,
+                            private val vilkårsvurderingRepository: VilkårsvurderingRepository,
+                            private val behandlingRepository: BehandlingRepository,
+                            private val søknadService: SøknadService,
+                            private val vedtakService: VedtakService,
+                            private val behandlinghistorikkService: BehandlingshistorikkService,
+                            private val tilkjentYtelseService: TilkjentYtelseService,
+                            private val fagsakService: FagsakService,
+                            private val persisterGrunnlagsdataService: PersisterGrunnlagsdataService) {
+
+    fun tilDto(behandling: Behandling, beslutter: String): IverksettDto {
+
+        val fagsak = fagsakService.hentFaksakForBehandling(behandling.id)
+        val vedtak = vedtakService.hentVedtak(behandling.id)
+        val forrigeBehandlingId = behandlingRepository.finnSisteIverksatteBehandling(behandling.id)
```

Comment:
Litt usikker på hele flytet med forrigeBehandlingId her. Hva skjer når forrige er teknisk opphør? 

Eks dette flytet
Førstegangsbehandling
Teknisk opphør
Ny behandling (forrigeBehandlingId burde være null her?)

forrigeBehandlingId blir brukt for å plukke opp andelene fra forrige behandlingId, sånn att vi kan beregne nytt utbetalingsoppdrag

Reply:
Vi har tolkat att forrigeBehandlingId  ska brukes til detta fæltet -> https://github.com/navikt/familie-eksterne-kontrakter/blob/master/stonadsstatistikk-ef/src/main/kotlin/no/nav/familie/eksterne/kontrakter/ef/BehandlingDVH.kt#L12 
Nytt utbetalingsoppdrag beregnes vel i ef-sak?

Reply:
Teknisk opphør er jo en slags revurdering med "0" i andelTilkjentYtelse. Så det blir jo på én måte riktig å bruke "teknisk opphør" fordi det er det siste vi har sendt videre til oppdrag/familie oppdrag. For i mellomtiden kan det jo ha skjedd ting i Infotrygd og vi har ikke "all historikk".

Reply:
På en måte er jo teknisk opphør en slags revurdering med kr 0 i andeler?